Update scraper

diogoscf opened the pull request:
Fixes #6 

- The main change is the addition of `scraper.py` in `scrapers` ~~(It must be run from the `scrapers` dir, or it will fail)~~
   - ~~This scraper, by default, gets the history of all languages and pairs ,as well as some other data used by the visualizer, for languages in the specified family, but it can also be run in "shallow" mode (`-s` or `--shallow`) to only get a bit of data. This shallow mode only updates `{family}_pairData.json` and `{family}_transducers.json`, while the normal mode updates all `{lang}.json` files as well. (To change the languages scraped, make sure to also change the `iso3to2` and `family` variables)~~

   - ~~Because of an issue with the Github API not working well with file renames, the script clones the repo uses submodules for each language and pair and gets the data from running `git log` in it, which is slow, but there is no other option to get the full history of a pair. To avoid being slower than necessary, the scraper ignores commits that are already in the files and only gets data for the ones that aren't. This scraper should be updated if either the GH API is updated to support history past file renames or if [the stats-service](https://github.com/apertium/apertium-stats-service/) is updated to support history (see [this issue](https://github.com/apertium/apertium-stats-service/issues/46)).~~
   - ~~The scraper also has an "update mailmap" mode (`-u` or `--updatemailmap`) that outputs the committers that aren't on the `.mailmap` file. This file exists so that when committers use different names for different commits, their contributions aren't counted separately for `transducers.json`. There is no need to use this option without shallow mode, as it only logs committers that would go in `transducers.json`~~

Most of the above text is outdated after many changes. Check the included README for actual info about the scraper

- Some changes were made to `familymap.html` to support the new data

- The json files were deleted and then entirely regenerated with `scraper.py`

- The scraper files were moved into a new `scraper/` directory, with the old svn ones going into the `scraper/svn-old/` dir

The end result for the visualizer using the new files is at https://diogoscf.github.io/family-visualizations/familymap.html

dolphingarlic commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,268 @@
+                location = topic[len("apertium-"):]
+                break
+
+        subprocess.call('git clone https://github.com/apertium/apertium-{}'.format(language), stderr=subprocess.DEVNULL)
```
This line gives me an error when run. Rather use os.system or [GitPython](https://github.com/gitpython-developers/GitPython)

dolphingarlic replied:
Actually just splitting the command into arguments like
```bash
subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)])
```
works as well

dolphingarlic commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,268 @@
+
+        subprocess.call('git clone https://github.com/apertium/apertium-{}'.format(language), stderr=subprocess.DEVNULL)
+        shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
+        lines = subprocess.check_output('git log --format="%aN" --follow apertium-{0}.{0}.lexc'.format(language), cwd='./apertium-{}/'.format(language)).decode('utf-8')
```
This also gives me errors. I think git doesn't work with subprocess in general.

(The errors I get are of the form `FileNotFoundError: [Errno 2] No such file or directory:`)

diogoscf replied:
~~Are you running the script from the `scrapers` directory? Because this is the type of errors that happen when run from another directory.
Also, obviously this one fails if `git clone` fails because it has nowhere to run the command~~

diogoscf replied:
Ah no, this seems to be a Linux issue. But doing it with a list should work

dolphingarlic commented on scrapers/.mailmap:
```diff
@@ -0,0 +1,50 @@
+Aidana <askaribayeva@gmail.com>
+Saule Zholdybekova <saule123@users.sourceforge.net>
+Zhandos Zhumanov <z.zhake@gmail.com>
+\u041d\u0443\u0440\u0431\u043e\u043b\u0430\u0442 \u0416\u0430\u043d\u0431\u043e\u0441\u0443\u043d\u043e\u0432 <nurbolat_03.93@mail.ru>
```
Is it possible to decode the Unicode entities when writing to this file? This should be Cyrillic text

diogoscf replied:
`git` doesn't recognize it if it's in Cyrillic

dolphingarlic commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,268 @@
+            contributors.append({"user": contributor, "value": count})
+
+        wikiData = requests.get(wikiURL).text
+        rows = etree.fromstring(wikiData)[1][0][0][0][3][3][8]
```
This like gives me `xml.etree.ElementTree.ParseError: not well-formed (invalid token): line 5, column 18`

diogoscf replied:
I can't replicate this error... It works flawlessly for me, but maybe I have a fix for it

dolphingarlic replied:
[Here's the error I get](https://imgur.com/a/7auYbMY), even with the updated code. I'm running this on Ubuntu 18.10. Also, could you change all the double quotes to single quotes or vice versa for consistency please?

diogoscf replied:
Tested it on a VM with Ubuntu 18 and still can’t replicate the error. I did run the code after all of my recent changes, so maybe I accidentally fixed it?

sushain97 commented on familymap.html:
```diff
@@ -171,36 +171,12 @@
-                {"langs": ["tat", "rus"], "location": "incubator", "stems": 5992},
-                {"langs": ["chv", "rus"], "location": "incubator", "stems": 75}
-            ];
+                githubRoot = "https://github.com/apertium/"
```
Missing semicolon.

sushain97 commented on familymap.html:
```diff
@@ -681,7 +636,7 @@
 
                         var splitLocation = langData.location.split(" "),
-                            svnLink = svnRoot + splitLocation[1].substring(1, splitLocation[1].length - 1) + "/" + splitLocation[0];
+                            svnLink = githubRoot + splitLocation[0]
```
```suggestion
                            svnLink = githubRoot + splitLocation[0];
```

sushain97 commented on familymap.html:
```diff
@@ -1032,7 +988,11 @@
+                  transducers = data;
+                  createChart();
+                })
+
```
```suggestion
```

sushain97 commented on familymap.html:
```diff
@@ -1032,7 +988,11 @@
 
             $(document).ready(function () {
-                createChart();
+                d3.json("./json/transducers.json", function(error, data){
```
```suggestion
                d3.json("./json/transducers.json", function(error, data) {
```

sushain97 commented on scrapers/dixcounter.py:
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python3
```
Did this file come another github repo? If so, can we use a submodule?

diogoscf replied:
No, I took it from the svn repo

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+
+from collections import Counter
+import xml.etree.ElementTree as etree
+import argparse, re, sys, logging, os, json, requests, subprocess, shutil, stat
```
These should be one per line.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+from collections import Counter
+import xml.etree.ElementTree as etree
+import argparse, re, sys, logging, os, json, requests, subprocess, shutil, stat
+from lexccounter import countStems as countLexcStems
```
Newline between stdlib and relative imports is conventional.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+    'tyv',
+    'uig',
+    'uzb'
+    ]
```
Indentation error. Suggest just running all this code through Black or similar.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+    'tur': 'tr',
+    'uig': 'ug',
+    'uzb': 'uz'
+    }
```
```suggestion
}
```

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+
+def del_rw(action, name, exc):
+    """Force deletes file if access is denied"""
+    os.chmod(name, stat.S_IWRITE)
```
Why is this necessary...? Why `shutil.rmtree` should work fine.

diogoscf replied:
I kept getting permission errors

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+def monoHistory(language):
+    """Returns the history of a monolingual dictionary"""
+    try:
+        oldFile = json.loads(open("../json/{}.json".format(language), "r").read())
```
```suggestion
        oldFile = json.load("../json/{}.json".format(language))
```

sushain97 replied:
This file could be easily improved to not require being run from this directory by using the location of itself instead of relative cwd paths.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+    except FileNotFoundError:
+        history = []
+
+    subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)], stderr=subprocess.DEVNULL)
```
Redirecting stderr to /dev/null will likely confuse someone in the future when this command fails and they don't see any error output. Can you make `git` quiet instead? What was the reason here?

diogoscf replied:
For some annoying reason, git logs the `Cloning into x...` as stderr, so I just made it silent

sushain97 replied:
`--quiet` is far better than completing chucking its output.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+        history = []
+
+    subprocess.call(['git', 'clone', 'https://github.com/apertium/apertium-{}'.format(language)], stderr=subprocess.DEVNULL)
+    shutil.copyfile("./.mailmap", "./apertium-{}/.mailmap".format(language))
```
'apertium-{}'.format(language)` could probably stand to be its own variable at this point.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+            continue
+        commitData = {}
+        lines = commit.split('\n')
+        for line in lines:
```
This is some pretty involved parsing here. Can you use `--format` to format things in a way that's more straightforward to parse and less error-prone?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+            continue
+
+        lexcFile = requests.get("https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.lexc".format(language, commitData["sha"])).text
+        if lexcFile == "404: Not Found\n":
```
It should be possible to check the response status instead. I wouldn't rely on this text.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+        if lexcFile == "404: Not Found\n":
+            lexcFile = requests.get("https://raw.githubusercontent.com/apertium/apertium-{0}/{1}/apertium-{0}.{0}.lexc".format(iso3to2[language], commitData["sha"])).text
+        try:
+            sys.stdout = open(os.devnull, "w")
```
Can you just silence the script instead? This is pretty hacky :)

sushain97 replied:
Maybe via setting the log output level.

diogoscf replied:
It's actually printing not logging, so I can only silence it this way (that I know of at least)

sushain97 replied:
Is it not possible to change the script?

diogoscf replied:
Yes. I am stupid sometimes

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+        if not set(pairList) <= set(languages):
+            continue
+
+        dixName = pairName if pairName != "tat-kir" else "tt-ky"
```
Please explain this via a comment. It's pretty special-case'y.

sushain97 replied:
(or ideally just use `iso3to2` to do it?)

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+        shutil.rmtree("./apertium-{}/".format(pairName), onerror=del_rw)
+
+        try:
+            oldFile = json.loads(open("../json/{}.json".format(language), "r").read())
```
Similar note here as above.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+        except FileNotFoundError:
+            history = []
+
+        for commit in commits:
```
Similar note as above for the parsing logic. Also seems like it could be DRY'd up?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+                dixFile = "https://raw.githubusercontent.com/apertium/apertium-{0}/{2}/apertium-{1}.{1}.dix".format(pairName, dixName, commitData["sha"])
+                stems = countDixStems(dixFile, True)
+                if stems == -1:
+                    continue
```
I'd almost suggest logging something for these exceptional cases. Perhaps a warning. Scripts don't need to be completely silent!

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+            mailmap = open("./.mailmap", "r", encoding="utf-8").read()
+            for email in emails:
+                if not email in mailmap:
+                    print(email)
```
?

diogoscf replied:
If run with the `--updatemailmap` option, the script outputs all committer emails that aren't in the `.mailmap` file.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+                if not email in mailmap:
+                    print(email)
+
+        shutil.rmtree("./apertium-{}/".format(language), onerror=del_rw)
```
Were you actually running into situations where you needed to mess with permission bits?

diogoscf replied:
Yes, I kept getting Access Denied

sushain97 replied:
Are you running the script as yourself or root? Maybe you're mixing WSL and Windows somehow..?

diogoscf replied:
I don't use WSL, but rather have installed pretty much everything I need. I've tried running git bash as admin, but that doesn't work either. I don't rly understand why, but this workaround fixes it

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+            if topic[len("apertium-"):] in pairLocations:
+                location = topic[len("apertium-"):]
+                break
+        stems = json.loads(requests.get("https://apertium.projectjj.com/stats-service/apertium-{}/bidix".format(pairName)).text)["stats"][0]["value"]
```
Why bother using stats-service if it won't give you the history?

diogoscf replied:
Right here, it only needs the current amount of stems, so it's much easier to use the already calculated ones from stats-service

sushain97 replied:
Hmm, okay. Are there different modes to get latest vs update history?

diogoscf replied:
Yes, check my original comment on this PR

sushain97 replied:
gotcha

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,267 @@
+            langHistory = []
+            langHistory.append(monoHistory(lang))
+            langHistory.extend(pairHistory(lang, allPackages))
+            outputFile = open("../json/{}.json".format(lang), "w+", encoding="utf-8")
```
Do we really want to append to the files here and above?

diogoscf replied:
Why not?

sushain97 replied:
Because JSON appended to JSON isn't valid JSON?

diogoscf replied:
Ah, but this isn't appending. It's overwriting each file. That why it's opened with the `w` argument. To append, it has to be opened with the `a` arg

sushain97 replied:
Gotcha. I mixed up my file modes. `+` meant append in my head.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+    dirName = "apertium-{}".format(language)
+    try:
+        oldFile = json.load(
+            open(os.path.join(path, "json", "{}.json".format(language)), "r")
```
Don't need the `open` here.

diogoscf replied:
You do actually. According to the JSON library documentation the given argument must have a `read()` property, and I was getting errors when I wasn't using `open`

diogoscf replied:
You do actually. According to the JSON library documentation the given argument must have a `read()` property, and I was getting errors when I wasn't using `open`

diogoscf replied:
You do actually. According to the JSON library documentation the given argument must have a `read()` property, and I was getting errors when I wasn't using `open`

sushain97 replied:
Ah, yeah, I was thinking about the `read` which you've since removed. 👍

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                history = data["history"]
+                break
+            history = []
+    except (FileNotFoundError, json.decoder.JSONDecodeError):
```
Why would there be a JSON decode error?

diogoscf replied:
For some reason I don't fully understand sometimes open() doesn't throw a FileNotFoundError, so JSON then throws a decoder error because it's trying to load nothing

sushain97 replied:
Is this deterministic? It seems more likely that the file exists and is empty or something of that nature.

diogoscf replied:
Possible, but it's still good that this catches it

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+            history = []
+    except (FileNotFoundError, json.decoder.JSONDecodeError):
+        history = []
+    if not Path(os.path.join(path, "scrapers", dirName)).exists():
```
But now it won't get updated, right?

diogoscf replied:
It deletes the repos at the end of the script, so this is just to avoid cloning the same repo multiple times

sushain97 replied:
I'd like it to keep the repos around so they can be updated instead of being cloned from scratch on a re-run.

diogoscf replied:
Ah yes, that's smarter.

diogoscf replied:
Should I add them as submodules?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                "--quiet",
+                "https://github.com/apertium/{}".format(dirName),
+            ],
+            cwd=os.path.join(path, "scrapers"),
```
It seems like you could save this path as a constant somewhere and use it everywhere?

sushain97 replied:
If you used `Path`, I think the resulting code would be even nicer! e.g. `SCRAPERS_DIR / '.mailmap'`.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+    os.remove(name)
+
+
+def monoHistory(language, path):
```
If `path` here is never going to change, would it make more sense as a constant at the top of the file?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+
+    commits.pop()  # last line is always empty
+    for commit in commits:
+        commitData = {}
```
I'd just declare this below as `commitData = {'sha': data[0], ...}`

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+        wikiData = requests.get(wikiURL).text
+        rows = etree.fromstring(wikiData, parser=etree.XMLParser(encoding="utf-8"))[1][
+            0
+        ][0][0][3][3][8]
```
This is really gnarly. Is it possible to use XPath instead?

sushain97 replied:
Or, at least, a comment?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+    for package in packages:
+        if not (
+            re.match(r"apertium-\w+$", package["name"])
+            and package["name"][len("apertium-") :] in languages
```
We do this `len("apertium-") :` thing a bunch of times. Could you move it into a function?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+        stats = json.loads(
+            requests.get(
+                "https://apertium.projectjj.com/stats-service/{}/Lexc".format(dirName)
+            ).text
```
`.json()` should work.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                location = topic[len("apertium-") :]
+                break
+
+        if not Path(os.path.join(path, "scrapers", dirName)).exists():
```
Looks like there are 3 places with this code. Can it be DRY'd up a bit?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                "https://apertium.projectjj.com/stats-service/apertium-{}/bidix".format(
+                    pairName
+                )
+            ).text
```
`.json()`.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                    pairName
+                )
+            ).text
+        )["stats"][0]["value"]
```
Does getting the `0`th stat always work?

diogoscf replied:
Yes. It return a single element list

sushain97 replied:
stats-service can return as many stats as it wants though. It just happens to return one for this particular file-type right now: https://github.com/apertium/apertium-stats-service/blob/2cde9217598a19d5e13ac0deb72fe872b19a7874/src/models.rs#L11. I'd suggest making this code robust to the service returning more things.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+
+    logging.getLogger("countStems").setLevel(logging.CRITICAL)
+
+    dirPath = Path(__file__).absolute().parent.parent
```
I'd put this at the top as a constant to avoid constantly passing it around.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+    dirPath = Path(__file__).absolute().parent.parent
+
+    allPackages = json.loads(
+        requests.get("https://apertium.projectjj.com/stats-service/packages").text
```
`.json()`

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+                encoding="utf-8",
+            )
+            json.dump(langHistory, outputFile, ensure_ascii=False)
+    for dir in next(os.walk(os.path.join(dirPath, "scrapers")))[1]:
```
Please avoid shadowing builtins.

sushain97 commented on scrapers/scraper.py:
```diff
@@ -0,0 +1,434 @@
+            )
+            json.dump(langHistory, outputFile, ensure_ascii=False)
+    for dir in next(os.walk(os.path.join(dirPath, "scrapers")))[1]:
+        if "apertium-" in dir:
```
Does this delete all the git repos? How does this work with the not-cloning unless present?

diogoscf commented:
I don't have time to finish up the last few changes now, but can you already check the changes, @sushain97 ?

diogoscf commented:
Ok all done. I squashed my commits, but left my last two separate (as the "More changes" commit) so it's clear what I changed

sushain97 commented:
Seems fine to me in terms of the code. @jonorthwash do you want to try it out to confirm it works for you?

diogoscf commented:
Okay @jonorthwash this should be easier to understand with the `README.md`

diogoscf commented:
Now scrapes according to a specified family. Also added better documentation

jonorthwash commented:
@sushain97, I'm not sure I understand the insistence on having (all the apertium??) repos as subrepos of the script.  I agree that they shouldn't be deleted after the script runs, but couldn't they just be cloned?  Why do they have subrepos?

sushain97 commented:
I didn't actually want this implementation. I preferred them just being cloned on initial run and just updated on later runs. @diogoscf went for the submodule approach and I couldn't see why that wouldn't work if implemented correctly 🤷‍♂

diogoscf commented:
Oh so I totally misunderstood. I thought you wanted the pushed submodules approach. Since @jonorthwash prefers it without submodules, I'll revert back (without deleting them in later runs obviously)

jonorthwash commented:
The problem with the submodule approach is that it makes this repo really big for no reason.  It's easy to just clear up cloned repos and have the script reclone them as needed.

jonorthwash commented:
```bash
$ ./scraper.py indic
Getting pair data for indic...
Getting monolingual data for indic...
Traceback (most recent call last):
  File "./scraper.py", line 451, in <module>
    monoData(allPackages, langs, family, args.updatemailmap),
  File "./scraper.py", line 245, in monoData
    prepRepo(dirName)
  File "./scraper.py", line 46, in prepRepo
    ["git", "pull", "--force", "--quiet",], cwd=repo,
  File "/usr/lib/python3.7/subprocess.py", line 339, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/usr/lib/python3.7/subprocess.py", line 800, in __init__
    restore_signals, start_new_session)
  File "/usr/lib/python3.7/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'apertium-ben': 'apertium-ben'
```

jonorthwash commented:
We also need status messages when it clones, even if it's a single message for cloning a whole bunch of repos.

sushain97 commented:
> The problem with the submodule approach is that it makes this repo really big for no reason. It's easy to just clear up cloned repos and have the script reclone them as needed.

This repo itself isn't particularly big until you actually clone the submodules which is the same as having the script run. By default, nothing is actually cloned.

jonorthwash commented:
> > The problem with the submodule approach is that it makes this repo really big for no reason. It's easy to just clear up cloned repos and have the script reclone them as needed.
> 
> This repo itself isn't particularly big until you actually clone the submodules which is the same as having the script run. By default, nothing is actually cloned.

True, but it seems really weird to have a script's dependency be a whole bunch of subrepos in its repo.  It also seems an annoyance to keep up to date (adding new repos all the time, etc.).  Anyway, @diogoscf says he switched to the other approach, so it doesn't matter much anymore.

sushain97 commented:
> True, but it seems really weird to have a script's dependency be a whole bunch of subrepos in its repo. It also seems an annoyance to keep up to date (adding new repos all the time, etc.). Anyway, @diogoscf says he switched to the other approach, so it doesn't matter much anymore.

Yep, I agree that it was odd :)

diogoscf commented:
There are now slightly better messages, I've fixed the error @jonorthwash was getting and I also added a `.gitignore` as I feel it was now a good idea

diogoscf commented:
Any reason this isn't merged yet?

sushain97 commented on scrapers/scraper.py:
```diff
@@ -440,15 +443,21 @@ def pairData(packages, languages):
         JSON_DIR.joinpath("{}_pairData.json".format(family)), "w+", encoding="utf-8",
     )
     if not args.quiet:
```
This would be best implemented as setting the logging level and then using `logging.info` or `logging.debug` everywhere.

diogoscf replied:
There is one problem with this, as there are warnings set for when `lexccounter` or `dixcounter` fail to count the stems for some reason, that can be turned on with `-v`. I don't think it would make sense to set this as a lower logging level that those infos, but then there would be no way of outputting that info without outputting those warnings

diogoscf replied:
Actually I could make it into debug. Would also make sense

sushain97 replied:
IIRC, you can set log levels on a per-logger basis.

diogoscf replied:
True, but come to think of it, it makes more sense as a `debug`, so it's fine either way

diogoscf replied:
Since you most likely couldn't solve the issue anyways, it doesn't really make sense as a warning

diogoscf commented:
I also added a better message for when the script fails because of an invalid language in the `families.json` file

sushain97 commented:
Code LGTM. @jonorthwash ?

jonorthwash commented:
Is there a reason a lot of the graphs start at non-zero numbers?  Some even seem to start in the thousands of stems...

diogoscf commented:
@jonortwash Some monolingual language dictionaries were copied from a pair repo and there is no way to track that. And if it's not a very large number, it could also be that the first commit already included some stems

diogoscf commented:
Though I might have a way to make it a bit better. Shouldn't be too much work

diogoscf commented:
Ok this is a bit better as it includes tracking past file renames even if the rename wasn't related to iso standards (e.g. the turkish monolingual dictionary used to be called `turmorph.lexc`) @jonorthwash

diogoscf commented:
So is this good now? @jonorthwash

jonorthwash commented:
I haven't tested the latest version yet, but we definitely want this code merged, so I suppose any further issues can be dealt with later.
